.github/workflows/LINUX_BUILD_TEST.yml

```diff
@@ -0,0 +1,736 @@
+name: Linux
+
+on:
+  workflow_call:
+    inputs:
+      SIMENG-MODE:
+        required: true
+        type: string
+      RUNNER:
+        default: ubuntu-latest
+        required: true
+        type: string
+
+env:
+  ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
+  LLVM-VERSION: 14
+  BENCHMARK_BRANCH: 'make-file-build-system' # The branch inside the benchmark repo that has the script to run all benchmarks.
+  PAT: ${{ secrets.SIMENGUOB_PAT }}
+
+
+jobs:
+  Build_and_Run:
+    runs-on: ${{ inputs.RUNNER }}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        
+        COMPILER: ['gcc-7', 'gcc-8', 'gcc-9', 'gcc-10'] # todo 'armclang'] # compiler names
+        OS: ['ubuntu:18.04','ubuntu:20.04', 'rockylinux:8', 'redhat/ubi8:latest', 'redhat/ubi9:latest', 'debian:10', 'debian:11'] # Docker images
+        
+        #######################################
+        # Removes unecessary jobs as jobs are generated in the order seen in the matrix.
+        # "Exclude" is to keep job ordering nice i.e. keeping ubuntu jobs next to each other in the list.
```

Comment:
Currently 12 are excluded, 14 are included. It _would_ be a bit cleaner to specify inclusions, and that'd mean the `select_setup` action could be cleaned to a single case where it takes in `OS` and `Compiler` and always works. That may even work atm, depending on where the following actually get excluded.

Tradeoff between time to rework vs how much it actually matters though, as current implementation seems clean enough

Reply:
Yeah, could change but doesn't really make much of a difference.  So not going to change here.